mesonbuild/compilers/mixins/clang.py

```diff
@@ -108,6 +108,12 @@ def openmp_flags(self) -> T.List[str]:
         else:
             # Shouldn't work, but it'll be checked explicitly in the OpenMP dependency.
             return []
+    
+    def get_win_subsystem_args(self, value: str) -> T.List[str]:
+        if self.info.is_windows() and not self.info.is_cygwin() and isinstance(self.linker, ClangClDynamicLinker):
+            return [f'-Wl,/subsystem:{value}']
+        else:
+            return super().get_win_subsystem_args(value)
```

Comment:
Sigh, this really should be a linker property, it would "just work" in that case. But it's not your fault.

Can you drop the `else:` after the return statement, ie:
```python
if condition:
    return something
return other_thing
```

Reply:
> I've got one style nit, and one "Why haven't I personally done this giant refactor I've been meaning to do" comment to myself you can ignore :)
> 
> Otherwise this looks good. The macOS failure isn't yours, that looks broken in master.

If there was a guiding issue outlining a plan for a linker related refactoring I'd be interested in contributing if I can.

Reply:
I did a refactor a while back that split the compilers and linkers into two objects, and then composes them. What we really need though is a true split, and probably a linker driver representation. I've got a WIP branch. The biggest problem is going to be getting it working with some of the obscure and/or closed source compilers.